app/controllers/recipes_controller.rb

```diff
@@ -0,0 +1,17 @@
+class RecipesController < ApplicationController
+
+  def show
+    @recipe = Recipe.find(params[:id])
+    @ingredients = @recipe.ingredient_names
+    @cost = @recipe.ingredients_cost
```

Comment:
Once again, you're adhering to MVC very well. For this case, though, having @ingredients and @cost instance variables is not strictly necessary. While we don't want any AR logic in the view, it's okay to call @recipe.ingredient_names or @cost.ingredients_cost from the view and just pass the one @recipe variable to it. This is just for the future if you want to minimize instance variables. Just another way to do it!